src/ovs/ovs_control.cpp

```diff
@@ -100,654 +100,615 @@ bool OVS_Control::bundle;
-            if (vconn_get_version(vconn) == OFP10_VERSION) {
-                set_packet_in_format(vconn, OFPUTIL_PACKET_IN_NXT, false);
-            }
-        }
+      if (vconn_get_version(vconn) == OFP10_VERSION) {
+        set_packet_in_format(vconn, OFPUTIL_PACKET_IN_NXT, false);
+      }
     }
-    
-    monitor_vconn(vconn, true, resume_continuations, bridge);
+  }
+
+  monitor_vconn(vconn, true, resume_continuations, bridge);
+}
+
+void OVS_Control::packet_out(const char *bridge, const char *options)
+{
+  enum ofputil_protocol usable_protocols;
+  enum ofputil_protocol protocol;
+  struct ofputil_packet_out po;
+  struct vconn *vconn;
+  struct ofpbuf *opo;
+  char *error;
+
+  error = parse_ofp_packet_out_str(&po, options, ports_to_accept(bridge),
+                                   tables_to_accept(bridge), &usable_protocols);
+  if (error) {
+    //ovs_fatal(0, "%s", error);
+    ACA_LOG_ERROR("%s", error);
+  }
+  protocol = open_vconn_for_flow_mod(bridge, &vconn, usable_protocols);
+  opo = ofputil_encode_packet_out(&po, protocol);
+  transact_noreply(vconn, opo);
+  vconn_close(vconn);
+  free(CONST_CAST(void *, po.packet));
+  free(po.ofpacts);
 }
 
-void
-OVS_Control::packet_out(const char *bridge, const char *options)
+int OVS_Control::dump_flows(const char *bridge, const char *flow, bool show_stats)
 {
-    enum ofputil_protocol usable_protocols;
+  ACA_LOG_DEBUG("%s", "OVS_Control::dump_flows ---> Entering\n");
+
+  int rc = EXIT_FAILURE;
+  int n_criteria = 0;
+
+  ACA_LOG_INFO("Executing dump_flows on bridge: %s, flow: %s, show_stats: %d\n",
+               bridge, flow, show_stats);
+
+  auto openflow_client_start = chrono::steady_clock::now();
+
+  if (!n_criteria && !should_show_names() && show_stats) {
+    dump_flows__(bridge, flow, false);
+    rc = EXIT_SUCCESS;
+  } else {
+    ofputil_flow_stats_request fsr;
     enum ofputil_protocol protocol;
-    struct ofputil_packet_out po;
     struct vconn *vconn;
-    struct ofpbuf *opo;
-    char *error;
 
-    error = parse_ofp_packet_out_str(&po, options,
-                                         ports_to_accept(bridge),
-                                         tables_to_accept(bridge),
-                                         &usable_protocols);
-    if (error) {
-        //ovs_fatal(0, "%s", error);
-        ACA_LOG_ERROR("%s", error);
+    vconn = prepare_dump_flows(bridge, flow, false, &fsr, &protocol);
+    struct ofputil_flow_stats *fses;
+    size_t n_fses;
+    run(vconn_dump_flows(vconn, &fsr, protocol, &fses, &n_fses), "dump flows");
+
+    struct ds s = DS_EMPTY_INITIALIZER;
+    for (size_t i = 0; i < n_fses; i++) {
+      ds_clear(&s);
+      ofputil_flow_stats_format(&s, &fses[i], ports_to_show(bridge), tables_to_show(bridge),
+                                //ports_to_show(ctx->argv[1]),
+                                //tables_to_show(ctx->argv[1]),
+                                show_stats);
+      printf(" %s\n", ds_cstr(&s));
```

Comment:
Should we change `printf` to `ACA_LOG_DEBUG` here?

Reply:
fixed.